hw1/02-palindrome.js

```diff
@@ -1,3 +1,25 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("input");
 
-elem.addEventListener('input', handleInput);
+elem.addEventListener("input", handleInput);
+
+function handleInput() {
+  const value = parseFloat(elem.value);
+  const resultDiv = document.querySelector("#result");
+
+  if (value >= 0) {
+    const numString = value.toString();
+    const reverseString = numString.split("").reverse().join("");
+
+    if (numString == reverseString) {
+      resultDiv.textContent = "Yes. This is a palindrome!";
+      document.getElementById("result").style.color = "green";
+    } else {
+      resultDiv.textContent = "No. Try again.";
+      document.getElementById("result").style.color = "red";
+    }
+  } else {
+    const resultDiv = document.querySelector("#result");
+    resultDiv.textContent = "Please enter a positive number.";
+    document.getElementById("result").style.color = "black";
```

Comment:
Just a style choice but consider making the error message for negative numbers red.

Reply:
can we not handle this by bootstrap using some defined class / ID?

Reply:
Yes, this can absolutely be handled using bootstrap (and should since it is the assignment requirement!). I was just pointing out that the message is just in black and would stand out more to the user if it was in red i.e. "text-warning" in bootstrap :)